runtime/src/accounts_db.rs

```diff
@@ -3243,6 +3243,147 @@ impl AccountsDb {
+            let slot = storages.first().unwrap().slot();
+            if slot == snapshot_slot {
+                continue;
+            }
+            if minimized_slot_set.contains(&slot) {
+                self.filter_storages(storages, minimized_account_set, &mut dead_storages);
+            } else {
+                dead_slots.push(slot);
+            }
+        }
+
+        self.purge_slots_from_cache_and_store(
+            dead_slots.iter(),
+            &self.external_purge_slots_stats,
+            true,
+        );
+        self.drop_or_recycle_stores(dead_storages);
+    }
+
+    fn get_minimized_slot_set(&self, minimized_account_set: &HashSet<Pubkey>) -> HashSet<Slot> {
+        let mut minimized_slot_set = HashSet::new();
+        self.thread_pool_clean.install(|| {
+            minimized_slot_set.par_extend(minimized_account_set.par_iter().flat_map(|pubkey| {
+                if let Some(read_entry) = self.accounts_index.get_account_read_entry(pubkey) {
+                    read_entry
+                        .slot_list()
+                        .iter()
+                        .map(|(slot, _)| *slot)
+                        .collect()
+                } else {
+                    HashSet::new()
+                }
+            }));
+        });
+        minimized_slot_set
+    }
+
+    fn filter_storages(
+        &self,
+        storages: Vec<Arc<AccountStorageEntry>>,
+        minimized_account_set: &HashSet<Pubkey>,
+        dead_storages: &mut Vec<Arc<AccountStorageEntry>>,
+    ) {
+        let slot = storages.first().unwrap().slot();
+        let (stored_accounts, _, _) = self.get_unique_accounts_from_storages(storages.iter());
+        let mut stored_accounts = stored_accounts.into_iter().collect::<Vec<_>>();
+        stored_accounts.sort_unstable_by(|a, b| a.0.cmp(&b.0));
+
+        let keep_accounts_collect = Mutex::new(Vec::with_capacity(stored_accounts.len()));
+        let remove_pubkeys_collect = Mutex::new(Vec::with_capacity(stored_accounts.len()));
+        let total_bytes_collect = AtomicUsize::new(0);
+        self.thread_pool_clean.install(|| {
+            const CHUNK_SIZE: usize = 50;
+            stored_accounts.par_chunks(CHUNK_SIZE).for_each(|chunk| {
+                let mut chunk_bytes = 0;
+                let mut keep_accounts = Vec::with_capacity(CHUNK_SIZE);
+                let mut remove_pubkeys = Vec::with_capacity(CHUNK_SIZE);
+                chunk.iter().for_each(|(pubkey, account)| {
+                    if minimized_account_set.contains(pubkey) {
+                        chunk_bytes += account.account_size;
+                        keep_accounts.push((pubkey, account));
+                    } else if let Some(read_entry) =
+                        self.accounts_index.get_account_read_entry(pubkey)
+                    {
+                        remove_pubkeys.push(pubkey);
+                        read_entry.unref();
```

Comment:
why unref here? Did this code get copy/pasted?

Reply:
Yeah, early on I thought this was necesssary and forgot to remove it. Thanks!